lib/amqp/client/connection.rb

```diff
@@ -416,12 +422,66 @@ def parse_frame(type, channel_id, buf) # rubocop:disable Metrics/MethodLength, M
         true
       end
 
+      def update_last_recv
+        return unless @heartbeat&.positive?
+
+        @last_recv = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      end
+
+      def update_last_sent
+        return unless @heartbeat&.positive?
+
+        @last_sent = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      end
+
       def handle_server_heartbeat(channel_id)
         return if channel_id.zero?
 
         raise Error::ConnectionClosed.new(501, "Heartbeat frame received on non-zero channel #{channel_id}")
       end
 
+      MAX_MISSED_HEARTBEATS = 2
+
+      # Start the heartbeat background thread (called from connection#tune)
+      def start_heartbeats(interval)
+        Thread.new do
+          loop do
```

Comment:
I think we want to set [`Thread#abort_on_exception=`](https://ruby-doc.org/3.4.1/Thread.html) to true like we do here? https://github.com/cloudamqp/amqp-client.rb/blob/00b521ca9bcd85dfb66baed3ad30e9cf3a90b2ba/lib/amqp/client.rb#L52-L56